### DISCUSSION TITLE
Transpiling from typescript to typescript

### DISCUSSION THREAD

**Discussion by quentinvernot** [babel/babel#14035] [Q&A] [answered]
Hello,

Is there a way to use babel to transpile typescript code _to_ typescript? I know, weird question, but it has been bugging me.

My goal here would be to use `preset-env` in the middle and ponyfill anything that my target needs with corejs, then still output typescript code. This would allow me to release a ts package where I don't need to document an endless, ever-changing, list of required polyfills for IE11 compatibility _by hand_, things would just work out of the box.

Doing this when transpiling to js is easy, but I have no idea how to do it with typescript.

Did anyone run into this?

Cheers!

**Comment by nicolo-ribaudo** [comment, marked as answer]
You can use `@babel/plugin-syntax-typescript` to enable TS parsing, rather than `@babel/plugin-transform-typescript` or `@babel/preset-typescript` which also transform it to JS.

**Reply by quentinvernot** [reply]
Hey, I forgot about the `syntax-` plugins!

Yeah that seems like it would work, thanks!

**Reply by cainrus** [reply]
> You can use `@babel/plugin-syntax-typescript` to enable TS parsing, rather than `@babel/plugin-transform-typescript` or `@babel/preset-typescript` which also transform it to JS.

is it possible to use babel plugins like `@babel/plugin-transform-destructuring`?
Right now plugins strips ts types after transformation. I want to use it with assemblyscript, to downgrade some neat modern syntax features to compatible with assemblyscript syntax.

**Reply by JLHwung** [reply]
> is it possible to use babel plugins like @babel/plugin-transform-destructuring?

Yes they should work in general, though preserving TS syntax while transforming other features is not as heavily tested as transforming both TS and JS features. Please file a bug report should you find any bugs.

**Reply by cainrus** [reply]
> Yes they should work in general, though preserving TS syntax while transforming other features is not as heavily tested as transforming both TS and JS features. Please file a bug report should you find any bugs.

Ok.
https://github.com/babel/babel/issues/14691
